In [438]:
import numpy as np
import pandas as pd
from stravalib.client import Client
import json

In [439]:
# Prepare data frames
df = pd.DataFrame(columns=['start','id','elapsed_time'])
df_temp = pd.DataFrame(columns=['start', 'id', 'elapsed_time'])

In [440]:
# Setup access to Strava API
with open('config.json') as data_file:    
    passwords = json.load(data_file)

client = Client()
access_token = passwords['passwords'][0]['access_token']
client_id = passwords['passwords'][1]['client_id']
client_secret = passwords['passwords'][2]['client_secret']

client.access_token = access_token

In [441]:
# Fetch activity data from Strava_API
for activity in client.get_activities():
    df_temp['start'] = [activity.start_date_local]
    df_temp['id'] = [activity.id]
    df_temp['elapsed_time'] = [activity.elapsed_time]
    df = df.append(df_temp, ignore_index=True)


In [450]:
# Analize data from Strava 
thresh_upper = np.timedelta64(100,"m")  # Upper bound in minutes
thresh_lower = np.timedelta64(30,"m")  # Lower bound in minutes

mask = df['elapsed_time'][20:].apply(lambda x: 
                                     True if (x > thresh_lower and x < thresh_upper) 
                                     else False)

filtered_array = df['elapsed_time'][20:].where(mask)

filtered_array.mean()

Timedelta('0 days 00:59:33.470238')

In [443]:
filtered_array.size - np.count_nonzero(np.isnat(filtered_array))

168

#Average commute time using CalTrain is 1:00 using 168 commutes for data